chunks/chunks.go

```diff
@@ -198,7 +198,44 @@ func (w *Writer) write(b []byte) error {
 	return err
 }
 
-func (w *Writer) WriteChunks(chks ...Meta) error {
+// mergeOverlappingChunks removes the samples whose timestamp is overlapping.
+// The first appearing sample is retained in case there is overlapping.
+func (w *Writer) mergeOverlappingChunks(chks []Meta) ([]Meta, error) {
+	if len(chks) < 2 {
+		return chks, nil
+	}
+	var newChks []Meta // Will contain the merged chunks.
```

Comment:
Can we allocate space for all of them?

`newChks := make([]Meta, len(chks), len(chks))`. It will reduce allocations. And it's fine if we will not use all of this. Statistically we will use all of those (when they not overlap).

Reply:
Thanks for this. I actually observed significant increase in allocs when I tested with https://github.com/prometheus/tsdb/pull/240. I will test again with this change.